tests/unit/test_charm.py

```diff
@@ -51,6 +53,29 @@ def test_synapse_pebble_layer(harness: Harness) -> None:
     }
 
 
+@pytest.mark.skip(reason="harness does not reproduce checks changes")
+def test_synapse_pebble_layer_change(harness: Harness) -> None:
+    """
+    arrange: charm deployed.
+    act: change experimental_alive_check config.
+    assert: Synapse charm should submit the correct Synapse pebble layer to pebble.
+    """
+    harness.set_leader(True)
+    harness.container_pebble_ready("synapse")
+    harness.begin_with_initial_hooks()
+    pebble_plan = harness.get_container_pebble_plan(synapse.SYNAPSE_CONTAINER_NAME).to_dict()
+    assert pebble_plan["checks"]["synapse-ready"]["period"] == "2m"
+    assert pebble_plan["checks"]["synapse-ready"]["threshold"] == 5
+    assert pebble_plan["checks"]["synapse-ready"]["timeout"] == "20s"
+
+    harness.update_config({"experimental_alive_check": "1m,3,30s"})
+
+    pebble_plan = harness.get_container_pebble_plan(synapse.SYNAPSE_CONTAINER_NAME).to_dict()
+    assert pebble_plan["checks"]["synapse-ready"]["period"] == "1m"
```

Comment:
should not this one be `synapse-alive`?

Reply:
I see the test is skipped for now

Reply:
At least until this one is in place :)
https://github.com/canonical/operator/pull/1268